### ISSUE TITLE
Read and apply custom build option on flash page

### ISSUE THREAD

**Issue by mituritsyn** [betaflight/betaflight-configurator#4228]
### Is your feature request related to a problem? Please describe

not related to problem

### Describe the solution you'd like

At current BFC will read and apply build options used in the firmware but will ignore custom defines

### Describe alternatives you've considered

parse and apply custom defines too
for example, having following [config](https://build.betaflight.com/api/builds/cf2ae0fbd2b1936c57638369ac13e2d0/json) after reflash I'll lost custom define for POS_HOLD_OPTION

https://build.betaflight.com/api/builds/f47415a1ce059c7d9b65b7d620e02351/log

**Comment by haslinghuis** [comment]
Already planned to add altitude hold and position hold to firmware build options.

**Comment by mituritsyn** [comment]
> Already planned to add altitude hold and position hold to firmware build options in [betaflight/betaflight#13974](https://github.com/betaflight/betaflight/pull/13974) but position hold is still in progress. There are a lot of custom defines possible.

poshold was just an example there always will be a bunch of other custom options

**Comment by nerdCopter** [comment]
agree, would be nice if "auto-detect" populated the "custom defines".

**Comment by haslinghuis** [comment]
- Mainstream `custom defines` will be added as build options in `Other Options` when production ready.
- As we no longer check build API but firmware defines directly - custom defines are not stored in firmware.

**Comment by haslinghuis** [comment]
Closing as this would need firmware and cloud API changes - we already add popular choices to Other Options and build options when applicable.
